### PR TITLE
[#174950220] Update and rebuild on remote changes

### PR DESCRIPTION
**Pull request by BurnedMarshal**
When an update happens on any remote repository used by `io-mock`, the command `yarn start` looks code changes and update the code inside the docker images.

### CODE REVIEW

**Comment by gunzip** [comment]
I'm not super entuxhiast about this approach (which is hard to maintain). I'd rather pre-process .env file (probably with a nodejs script) replacing occurences of 'master' with the hash commit:

`git fetch && git log -n 1 origin/master --pretty=format:"%H"`

for every run of `yarn start`

**Comment by BurnedMarshal** [comment]
> I'm not super entuxhiast about this approach (which is hard to maintain). I'd rather pre-process .env file (probably with a nodejs script) replacing occurences of 'master' with the hash commit:
> 
> `git fetch && git log -n 1 origin/master --pretty=format:"%H"`
> 
> for every run of `yarn start`

Ok, I try this approach

**Comment by BurnedMarshal** [comment]
> > I'm not super entuxhiast about this approach (which is hard to maintain). I'd rather pre-process .env file (probably with a nodejs script) replacing occurences of 'master' with the hash commit:
> > `git fetch && git log -n 1 origin/master --pretty=format:"%H"`
> > for every run of `yarn start`
> 
> Ok, I try this approach

@gunzip for doing this we must get the commit ID outside the repo (that exists only inside the container)

**Comment by gunzip** [comment]
true! can you try using 

curl -u "u:${GITHUB_TOKEN}" https://api.github.com/repos/pagopa/io-backend/git/ref/heads/master | jq .object.sha | tr -d '"' )

?

File: .env-example [hunk 1]
```diff
@@ -1,38 +1,47 @@
 API_GATEWAY_PORT=80
 COSMOSDB_PORT=3000
 
+FUNCTIONS_ADMIN_REPO=https://api.github.com/repos/pagopa/io-functions-admin/git/ref/heads/
 FUNCTIONS_ADMIN_CONTEXT=./io-functions-admin
 FUNCTIONS_ADMIN_BRANCH=master
```
Comment (gunzip) [inline .env-example:6]: are these BRANCH variables still used?

File: bin/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const dotenv = require("dotenv");
+const envsub = require("envsub");
+const fetch = require("node-fetch");
+dotenv.config();
+
+const [_, __, repoUrlEnv, repoBranchEnv, templatePath, destinationPath] = process.argv;
+(async () => {
+  const commitSha = await fetch(`${process.env[repoUrlEnv]}${process.env[repoBranchEnv]}`)
```
Comment (gunzip) [inline bin/index.js:8]: can you add the ability to use a GITHUB_TOKEN (ie. the one provided into .env) to make these calls? (if present). you're limited to 60 unauthenticaed calls for 1h and 5k using the token.

File: .env-example (hunk 1 above)
Comment (BurnedMarshal) [reply .env-example:6]: Yes, they are used to get the commit sha in [here](https://github.com/pagopa/io-mock/pull/18/files#diff-9ea21889e9fbc7d82e7dabb0fb280808R8)

File: bin/index.js (hunk 2 above)
Comment (BurnedMarshal) [reply bin/index.js]: Fixed in 5a9dcf0

**Review by gunzip — APPROVED** [review]

**Pull request merged**